src/NHibernate/Mapping/ByCode/ModelMapper.cs

```diff
@@ -1327,19 +1327,19 @@ private IMapKeyRelationMapper DetermineMapKeyRelationType(MemberInfo member, Pro
 
 		private class ComponentRelationMapper : ICollectionElementRelationMapper
 		{
-			private readonly MemberInfo collectionMember;
+			private readonly PropertyPath propertyPath;
 			private readonly System.Type componentType;
 			private readonly ICustomizersHolder customizersHolder;
 			private readonly IModelInspector domainInspector;
 			private readonly ICandidatePersistentMembersProvider membersProvider;
 			private readonly ModelMapper modelMapper;
 			private readonly System.Type ownerType;
 
-			public ComponentRelationMapper(MemberInfo collectionMember, System.Type ownerType, System.Type componentType, ICandidatePersistentMembersProvider membersProvider,
+			public ComponentRelationMapper(PropertyPath propertyPath, System.Type componentType, ICandidatePersistentMembersProvider membersProvider,
 			                               IModelInspector domainInspector, ICustomizersHolder customizersHolder, ModelMapper modelMapper)
 			{
-				this.collectionMember = collectionMember;
-				this.ownerType = ownerType;
+				this.propertyPath = propertyPath;
+				this.ownerType = propertyPath.LocalMember.DeclaringType;
```

Comment:
I don't believe it changes the behavior. Before the change, the `ComponentRelationParameter` constructor was being passed
- A `property` parameter of type `MemberInfo`
- An `ownerType` parameter of type 'System.Type'
I've replaced both of these parameters with
- A propertyPath parameter of type `PropertyPath`

I was under the impression that `PropertyPath` parameter already had the ownerType in the field `propertyPath.DeclaringMember.OwnerType`, so it was redundant to pass this in as a parameter.
There doesn't seem be a scenario where `propertyPath.DeclaringMember.OwnerType` differs from the `ownerType`.
Can you think of a possible scenario? If so, we should capture it in an automated test.

On another note, I'm trying to read the tests I wrote last October, and I think they're too hard to read. I refactored them too much. 

I should change them - introduce some duplication into each test, just so it's easier to follow through the steps that are being taken.